docs/installation/server-config/security.rst

```diff
@@ -132,28 +133,130 @@ Configuring Authorization of KSQL with Kafka ACLs
-:ref:`interactive <config-security-ksql-acl-interactive>` or :ref:`non-interactive (headless) <config-security-ksql-acl-headless>`.
+The list of ACLs that must be defined depends on the version of the Kafka cluster.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Kafka Cluster v2.0 and above
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Kafka 2.0 saw some enhancements that drastic simplifies the ACLs required to run KSQL against a Kafka cluster secured with ACLs.
+(See KIP-277 and KIP-290 for details). It is highly recommended to use Kafka versions of 2.0 or above for secure clusters.
+
+ACL definition
+^^^^^^^^^^^^^^
+
+Kafka ACLs are defined in the general format of "Principal P is [Allowed/Denied] Operation O From Host H on any Resource R matching ResourcePattern RP".
+
+Principal
+    An authenticated user or group. For example, ``"user: Fred"`` or ``"group: fraud"``.
 
-This section uses the terminology used by the :ref:`Kafka Authorizer <kafka_authorization>` (``SimpleAclAuthorizer``)
-to describe the required ACLs. Each ACL is made up of these parts:
+Permission
+    Defines if the ACL allows (``ALLOW``) or denies (``DENY``) access to the resource.
+
+Operation
+    The operation that is performed on the resource, for example ``READ``.
 
 Resource
     A resource is comprised of a resource type and resource name:
 
     - ``RESOURCE_TYPE``, for example ``TOPIC`` or consumer ``GROUP``.
-    - Resource name, where the name is either specific, e.g. ``users``, or the wildcard ``*``, meaning all resources of this type.
+    - Resource name, e.g. the name of a topic or a consumer-group.
 
-    The ``CLUSTER`` resource type does not require a resource name because it refers to the entire Kafka cluster.
+ResourcePattern
+    A resource pattern matches zero or more Resources and is comprised of a resource type, a resource name and a pattern type.
 
-Operation
-    The operation that is performed on the resource, for example ``READ``.
+    - ``RESOURCE_TYPE``, for example ``TOPIC`` or consumer ``GROUP``. The pattern will only match resources of the same resource type.
+    - Resource name. How the pattern uses the name to match Resources is dependant on the pattern type.
+    - ``PATTERN_TYPE``, controls how the pattern matches a Resource's name to the patterns. Valid values are:
+
+        - ``LITERAL`` pattern types match the name of a resource exactly, or, in the case of the special wildcard resource name `*`, resources of any name.
+        - ``PREFIXED`` pattern types match when the resource's name is prefixed with the pattern's name.
+
+    The ``CLUSTER`` resource type is implicitly a literal pattern with a constant name because it refers to the entire Kafka cluster.
+
+An literal example might be ``ALLOW`` ``user Fred`` to ``READ`` the ``TOPIC`` with the ``LITERAL`` name ``users``.
+
+An prefixed example might be ``ALLOW`` ``user Fred`` to ``READ`` any ``TOPIC`` whose name is ``PREFIXED`` with ``fraud-``.
```

Comment:
I'd rename Fred to Bob (to pick a new name and to make the examples more different) and reword to:

----

- A prefixed example: ``ALLOW`` ``user Bob`` to ``READ`` any ``TOPIC`` whose name is ``PREFIXED`` with ``fraud-``.  Here, user ``Bob`` would be allowed to read from topics such as ``fraud-europe``, ``fraud-testing``, ``fraud-1``, and ``fraud-``.  But ``Bob`` would not be allowed to read from topics such as ``production-fraud``.

----

Just an idea, because often examples work better than a grammar/BNF/...

Reply:
And I thought I was being verbose and spelling things out ;)